### ISSUE TITLE
Global ordering in interaction may get inconsistent

### ISSUE THREAD

**Issue by planger** [eclipsesource/papyrus-seqd#347]
The global ordering may become inconsistent with respect to message ends and ultimately depends on the order in which fragments have been added instead of their position they've been added in.

Consider for instance the following scenario:
![image](https://user-images.githubusercontent.com/588090/45291642-07759880-b4f3-11e8-8982-37af7d97edbf.png)

I've added them in the order `1`, `2`, and `3`. This is also their order in the interaction's global ordering (see Project Explorer). However, imho, this is wrong, as the send and receive events of message `3` (`3s`and `3r`) appear _after_ the send and receive event of message `1`and `2` (again see Project Explorer).

If I now add an execution on lifeline 2 before the message, you'll see that `1s` is before the added execution, while `1r` is after it (see Project Explorer). This is what causes the wrong nudging behavior described in https://github.com/eclipsesource/papyrus-seqd/issues/343#issuecomment-418695991.
![image](https://user-images.githubusercontent.com/588090/45292139-8fa86d80-b4f4-11e8-9510-db52b579d523.png)

Similarly, when I add executions on top of `Lifeline3`, only message `3` gets nudged down, also if all should be nudged down. You'll also see that a graphical/semantic inconsistency occurs, because all send and receive events remain before the added executions, except for `3r` (the receive event of message `3`), which again leads to unexpected nudging behavior: we only nudge down message `3` without nudging `1` and `2`, but we nudge the execution above (reported in https://github.com/eclipsesource/papyrus-seqd/issues/343#issuecomment-418695991).
![image](https://user-images.githubusercontent.com/588090/45292500-b6b36f00-b4f5-11e8-8800-c59cbe548255.png)

On a side-note: shouldn't we also keep the messages in order with their send- and receive-events? Currently the messages will always get appended. For executions, we are more consistent, as we first have the start occurrence, the execution, and then the end occurrence.

**Comment by cdamus** [comment]
Messages are unordered (as specified by `Interaction::message`), so appending them should be fine.

In PR #330 I've started fixing some interaction fragment ordering issues when creating messages (doing a better job of inserting the message ends in the correct slots in the `Interaction::fragment` list).

**Comment by planger** [comment]
Right, thanks a lot! #330 indeed fixes several of the issues, I documented in this issue (sorry, I had start to document those issues before looking at #330).
As far as I can see, the only issue that is remaining also after #330 is as follows.
Initial scenario:
![image](https://user-images.githubusercontent.com/588090/45299234-31d35000-b50b-11e8-9ec5-85d7c3a03844.png)

Now add an execution to the top of lifeline 3:
![image](https://user-images.githubusercontent.com/588090/45299260-4c0d2e00-b50b-11e8-885c-3a76345e87dd.png)

As can be seen above, the send event is still before the added execution, which leads to issue https://github.com/eclipsesource/papyrus-seqd/issues/343#issuecomment-418695991 as far as I've looked into it.
